data-pipeline/src/data_pipeline/datasets/clinvar.py

```diff
@@ -65,61 +69,71 @@ def _parse_submission(submission_element, trait_mapping_list_element):
 
         if trait_mapping_list_element is not None:
             xref_elements = trait_element.findall("XRef")
+
```

Comment:
I like the desire to introduce some explanation in here. One thing, though, is that when you end up writing a series of comments that walk you through a process step by step, it's a sign to consider refactoring by extracting helper functions to make the code more self-documenting.

Here I would specifically recommend extracting in two stages: in one, you break this whole big chonk of code out into a function called something like `find_mapping_elements` which takes `trait_element` as input and spits out the corresponding `preferred_name_element` and `mapping_element`. 

Then I'd extract each of the individual steps out of `find_mapping_elements` to something like:

```
def find_mapping_elements(trait_element, xref_elements):
  (preferred_name_element, mapping_element) = find_mapping_elements_by_explicit_xref(trait_element, xref_elements) ||
      find_mapping_elements_by_preferred_name(trait_element) ||
      find_mapping_elements_by_classification_name(trait_element)
```

And looking at it, everything from line 103 on is concerned with adding Medgen IDs rather than finding these mappings, so let's extract that stuff into its own set of functions.